AssistantAgent.metadata for user/application identity information associated with the agent. #6048

tongyu0924 opened the pull request:
<!-- Thank you for your contribution! Please review https://microsoft.github.io/autogen/docs/Contribute before opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR. If you don't have the access to it, we will shortly find a reviewer and assign them to your PR. -->

## Why are these changes needed?

This PR introduces a metadata field in AssistantAgentConfig, allowing applications to assign and track identity information for agents.
The metadata field is a Dict[str, str] and is included in the configuration for proper serialization.

## Related issue number

Closes #6048

## Checks

- [ ] I've included any doc changes needed for <https://microsoft.github.io/autogen/>. See <https://github.com/microsoft/autogen/blob/main/CONTRIBUTING.md> to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

ekzhu commented on python/packages/autogen-agentchat/src/autogen_agentchat/agents/_assistant_agent.py:
```diff
@@ -613,8 +614,10 @@ def __init__(
         reflect_on_tool_use: bool = False,
         tool_call_summary_format: str = "{result}",
         memory: Sequence[Memory] | None = None,
+        metadata: Dict[str, str] | None = None
```
We need to add this parameter to the API doc's argument list.

ekzhu reviewed (changes requested):
Looks good. Just the API docs need to be updated as well.

tongyu0924 commented:
> Looks good. Just the API docs need to be updated as well.


I've modified it, but I'm not sure if it's correct

ekzhu commented:
> > Looks good. Just the API docs need to be updated as well.
> 
> 
> 
> 
> 
> I've modified it, but I'm not sure if it's correct

It's not. The api doc is the doc string of AssistantAgent

tongyu0924 commented:
> > > Looks good. Just the API docs need to be updated as well.
> > 
> > 
> > I've modified it, but I'm not sure if it's correct
> 
> It's not. The api doc is the doc string of AssistantAgent

I've added the metadata docstring to AssistantAgent — could you help confirm if the wording is correct?

ekzhu commented:
Please run `poe format` to fix formatting issues.

tongyu0924 commented:
> Please run `poe format` to fix formatting issues.

I’ve run poe format and pushed the changes.
